lib/Ix/AccountState.pm

```diff
@@ -83,18 +83,15 @@ sub _save_states ($self) {
   my $rows = $self->_state_rows;
   my $pend = $self->_pending_states;
 
+  # XXX - Should only ever be one pending state. Croak if not true? Modify
+  #       code to expect only one? Other? -- alh, 2017-01-27
```

Comment:
Okay, so I'm looking at this code now, and here's the scenario I imagine:
* ix_set is called for setFoos
* set.create creates some Foos, then calls ensure_state_bumped(foos)
* …so now foos is pending
* set.update updates some Foos
* …so ix_updated is called, altering related Bars with a nested setBars
• …so ix_set on Bars calls  ensure_state_bumped(bars)
* …then when ix_set on Bars is about to return, it calls _save_state, and there are two pending states

In the case I originally thought of, where create A calls create B, we're safe because ix_created (which would do the nested create of B) would be called before A has called ensure_state_bumped.  The scenario I give above gets past that happy coincidence.